src/sql/workbench/services/objectExplorer/browser/serverTreeDataSource.ts

```diff
@@ -63,6 +63,8 @@ export class ServerTreeDataSource implements IDataSource {
 		} else if (element instanceof TreeNode) {
 			let node = element;
 			if (node.children) {
+				// Grab the latest data from the server of the node's children.
+				await this._objectExplorerService.refreshTreeNode(node.getSession()!, node);
```

Comment:
@alanrenmsft I had Alex move this here, since the root problem we discovered is that it was calling refreshTreeNode and waiting for it to finish before it ever called tree.refresh - which meant that the actual tree refresh was also super quick (since it just immediately got the nodes) but the spinner never showed up since it wasn't told it was refreshing. 

I haven't dug myself into how getChildren is used though so I'm worried there might be perf side effects from this. Can you think of anything this might cause problems with? (and if you have time to dig through the code a bit to see if there's anything we missed that'd be great...)

Reply:
let me dig into it a bit.